src/images/kafka.rs

```diff
@@ -1,44 +1,41 @@
-use crate::{core::WaitFor, Image};
+use crate::{core::WaitFor, Image, ImageArgs};
 use std::collections::HashMap;
 
 const NAME: &str = "confluentinc/cp-kafka";
-const DEFAULT_TAG: &str = "6.1.1";
+const TAG: &str = "6.1.1";
 
 pub const KAFKA_PORT: u16 = 9093;
 const ZOOKEEPER_PORT: u16 = 2181;
 
-#[derive(Clone, Debug, Default)]
+#[derive(Debug, Default, Clone)]
 pub struct KafkaArgs;
 
-impl IntoIterator for KafkaArgs {
-    type Item = String;
-    type IntoIter = ::std::vec::IntoIter<String>;
-
-    fn into_iter(self) -> Self::IntoIter {
-        vec![
-            "/bin/bash".to_owned(),
-            "-c".to_owned(),
-            format!(
-                r#"
+impl ImageArgs for KafkaArgs {
+    fn into_iter(self) -> Box<dyn Iterator<Item = String>> {
+        Box::new(
+            vec![
+                "/bin/bash".to_owned(),
+                "-c".to_owned(),
+                format!(
+                    r#"
 echo 'clientPort={}' > zookeeper.properties;
 echo 'dataDir=/var/lib/zookeeper/data' >> zookeeper.properties;
 echo 'dataLogDir=/var/lib/zookeeper/log' >> zookeeper.properties;
 zookeeper-server-start zookeeper.properties &
 . /etc/confluent/docker/bash-config &&
 /etc/confluent/docker/configure &&
 /etc/confluent/docker/launch"#,
-                ZOOKEEPER_PORT
-            ),
-        ]
-        .into_iter()
+                    ZOOKEEPER_PORT
+                ),
+            ]
+            .into_iter(),
+        )
     }
 }
 
 #[derive(Debug)]
 pub struct Kafka {
-    arguments: KafkaArgs,
     env_vars: HashMap<String, String>,
```

Comment:
I am kinda neutral (slightly leaning towards the latter) on whether we should have fields like these: here (or) just have them within `impl Image` blocks under respective methods and just `[derive(Default)]` here.

Reply:
I don't fully follow? Can you give an example of both?

Reply:
Sure. As you can see in https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f899fe85e2d4bb8a79d5f76b07f0e373 - why to define like `Foo` when can also define like `Bar` in the era of `RunnableImage` ?

Reply:
A fair enough yes. I am in favor of the latter as well.

Reply:
My bad that I didn't realize that we also need to change the return types of the methods `env_vars` and `volumes` in `Image` trait to `Box<dyn Iterator<Item = (String, String)>`. I'll think about this and we shall discuss more on this when I start to work on `ImageExt`. For now, I am leaving this as it is.